lib/controllers/calendar.dart

```diff
@@ -0,0 +1,332 @@
+import 'package:device_calendar/device_calendar.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_timezone/flutter_timezone.dart';
+import 'package:timezone/timezone.dart' as tz;
+
+import 'package:squadquest/common.dart';
+import 'package:squadquest/logger.dart';
+import 'package:squadquest/models/instance.dart';
+import 'package:squadquest/models/user.dart';
+
+/// {@template calendar_controller}
+///
+/// A controller for interacting with the device's calendar. This is used to create
+/// and update events in the device's calendar.
+///
+/// {@endtemplate}
+abstract interface class CalendarController {
+  static CalendarController? _instance;
```

Comment:
@IguJl15 I've been using Riverpod providers for singleton's instead of `.instance` members per https://codewithandrea.com/articles/flutter-singletons/#use-riverpod-providers

What are your thoughts?